subroutines.sh

```diff
@@ -3,39 +3,71 @@
 # subroutines.sh
 #
 
+
 # Function
 send_command() {
   vin=$1
-  shift
+  cmd=$2
+  cmd_name=$3
+
+  TESLA_CONTROL_CMD='/usr/bin/tesla-control -ble -vin $vin -key-name /share/tesla_blemqtt/${vin}_private.pem -key-file /share/tesla_ble_mqtt/${vin}_private.pem $cmd 2>&1'
 
+  # add a retry loop
   max_retries=5
-  for count in $(seq $max_retries); do
-    log_notice "Sending command $* to vin $vin, attempt $count/${max_retries}"
+  for tesla_ctrl_count in $(seq $max_retries); do
+
+    log_notice "Attempt $tesla_ctrl_count//${max_retries} sending $cmd_name to vin:$vin command:$cmd $2, attempt $tesla_ctrl_count/${max_retries}"
     set +e
     # shellcheck disable=SC2068
-    tesla_ctrl_out=$(tesla-control -vin $vin -ble -key-name /share/tesla_blemqtt/${vin}_private.pem -key-file /share/tesla_ble_mqtt/${vin}_private.pem $@ 2>&1)
+    tesla_ctrl_out=$(eval $TESLA_CONTROL_CMD)
     EXIT_STATUS=$?
     set -e
     if [ $EXIT_STATUS -eq 0 ]; then
-      log_info "tesla-control send command succeeded"
-      break
+      log_info "tesla-control command to vin:$vin was delivered"
+      return 0
```

Comment:
Again, we could mqtt back when a status update would be relevant (eg. climate on/off, charging state, ...) --> But maybe for next version

Reply:
Yes you rind in my mind... the other options would be to use pipes. Have a main loop that sends and reads from all background processes and nothing would block. Quickly in terms of "high availability" , 'm not sure which one is the best. MQTT we can lose messages is the service gets restarted. We can think about that one for upcoming release.